pkg/kubelet/sysctl/safe_sysctls.go

```diff
@@ -28,5 +28,6 @@ func SafeSysctlAllowlist() []string {
 		"net.ipv4.tcp_syncookies",
 		"net.ipv4.ping_group_range",
 		"net.ipv4.ip_unprivileged_port_start",
+		"net.ipv4.ip_local_reserved_ports",
```

Comment:
need an ack from sig-node / sig-network this is universally safe for containers to set (doesn't interfere with settings in other containers)

Reply:
https://www.kernel.org/doc/Documentation/networking/ip-sysctl.txt

```
ip_local_reserved_ports - list of comma separated ranges
	Specify the ports which are reserved for known third-party
	applications. These ports will not be used by automatic port
	assignments (e.g. when calling connect() or bind() with port
	number 0). Explicit port allocation behavior is unchanged.

	The format used for both input and output is a comma separated
	list of ranges (e.g. "1,2-4,10-10" for ports 1, 2, 3, 4 and
	10). Writing to the file will clear all previously reserved
	ports and update the current list with the one given in the
	input.

	Note that ip_local_port_range and ip_local_reserved_ports
	settings are independent and both are considered by the kernel
	when determining which ports are available for automatic port
	assignments.

	You can reserve ports which are not in the current
	ip_local_port_range, e.g.:

	$ cat /proc/sys/net/ipv4/ip_local_port_range
	32000	60999
	$ cat /proc/sys/net/ipv4/ip_local_reserved_ports
	8080,9148

	although this is redundant. However such a setting is useful
	if later the port range is changed to a value that will
	include the reserved ports.

	Default: Empty
```

it is related to `ip_local_port_range` that is already considered safe, @mrunalp @haircommander can you please double ACK it is safe?

Reply:
the question is whether this is a namespaced sysctl such that setting it for one container does not affect other containers, not whether the effect on a particular container is safe

Reply:
Yes, I believe it is namespaced. In my test:

```
$ docker run --rm --sysctl=net.ipv4.ip_local_reserved_ports=3967 -it busybox /bin/sh
/ # cat /proc/sys/net/ipv4/ip_local_reserved_ports
3967

# it is set in this container
```

```
# check if it set on the host => no it is not
$ cat /proc/sys/net/ipv4/ip_local_reserved_ports
# empty
```

```
# run another container attach to the network namespace of the one where the sysctl is set => results in sysctl being set
$ docker run --rm --net container:d4bceff75d35 busybox cat /proc/sys/net/ipv4/ip_local_reserved_ports
3967
```

```
# Verify in another container in another network namespace that it is not set
$ docker run --rm busybox cat /proc/sys/net/ipv4/ip_local_reserved_ports

# empty
```

Reply:
no sure how to check that. The documentation seems to be not changed.

- https://github.com/torvalds/linux/blob/v3.0/Documentation/networking/ip-sysctl.txt#L614
- https://github.com/torvalds/linux/blob/v6.1/Documentation/networking/ip-sysctl.rst

Reply:
https://github.com/torvalds/linux/commit/122ff243f5f104194750ecbc76d5946dd1eec934

It has been changed to namesapced since kernel v3.16 according to the commit information/

Reply:
Thanks for tracking that down. Is our minimum kernel version 3.16+?

Reply:
For kubelet side,
- probably, we should add kernel version check in before adding it to the safe sysctl list.

But for pod security, I am not sure which is the best practise. 
- We can add it to the list, and if kubelet cannot support it and kubelt will fail the pod laster.

Reply:
+1, thanks for digging up the kernel commit. It sounds reasonable to add a kernel check in kubelet then for this sysctl... v3.16 is pretty old (released in 2014), but we don't know what kernel versions folks will be using.

Reply:
LGTM

Reply:
I believe once we introduced 3.18 is the minimal kernel we plan to support. Let me dig into it to see if we enforce it upon the decision. Meanwhile I am ok to add a kernel check.